aea/helpers/async_friendly_queue.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the implementation of AsyncFriendlyQueue."""
+import asyncio
+import queue
+import threading
+from collections import deque
+from typing import Any
+
+
+class AsyncFriendlyQueue(queue.Queue):
+    """queue.Queue with async_get method."""
+
+    def __init__(self, *args, **kwargs):
+        """Init queue."""
+        super().__init__(*args, **kwargs)
+        self._lock = threading.Lock()  # to cover _waiters access
```

Comment:
is this needed? I haven't spotted usage

Reply:
@marcofavorito  thanks, lock usage was removed.
fixed